### PR TITLE
Add address check primitives

### PR DESCRIPTION
**Pull request by hanssv**
Added `Address.is_contract`, `Address.is_oracle`, `Oracle.check` and `Oracle.check_query`.

And a note on `Contract.creator`...

### CODE REVIEW

File: contracts/sophia.md [hunk 1]
```diff
@@ -798,7 +822,7 @@ The block-chain environment available to a contract is defined in three name spa
 `Contract`, `Call`, and `Chain`:
 
 - `Contract.creator` is the address of the entity that signed the contract creation
-  transaction.
+  transaction. (Available from Fortuna release.)
```
Comment (lucafavatella) [inline contracts/sophia.md]: Contract creation transaction only? Or also generalized accounts attach transaction?

Comment (hanssv) [reply contracts/sophia.md]: The attach is considered a contract creation in this aspect.

**Comment by hanssv** [comment]
contracts/aevm.md is horribly outdated, in need of a touch-up...

**Comment by hanssv** [comment]
> Where would I run this?

This the protocol repository it only has documentation...

**Pull request merged**